docs/components/SelectView.jsx

```diff
@@ -199,8 +216,21 @@ export default class SelectView extends Component {
             },
             {
               name: "options",
-              type: "Array",
-              description: "Possible options, must contain objects with label and value attributes.",
+              type: "{label: string; value: string}[]",
+              description: "Possible options, must contain objects with label and value attributes. Must not be set if `lazy`.",
+              optional: true,
+            },
+            {
+              name: "lazy",
+              type: "boolean",
+              description: "Set to true to fetch options asynchronously using the `loadOptions` function.",
+              defaultValue: "false",
+              optional: true,
+            },
+            {
+              name: "loadOptions",
+              type: "(input: string) => Promise<{options: {label: string; value: string}[]}>",
+              description: "Function to load options for a given search input. Required if `lazy`.",
```

Comment:
Personally, I think that behavior is really reasonable. Until we see an explicit need to configure it, I think we should leave it as-is (but I'll add the appropriate documentation). And even then, I think we should limit the configuration surface rather than just exposing a blanket `reactSelectOptions` prop.